### DISCUSSION TITLE
Insight object attribute

### DISCUSSION THREAD

**Discussion by oymelnyk** [dstengele/python-insight#12] [Q&A]
Hi, can you help me?
I want to get value of Insight object attribute.
Now I have Insight Object key, and I don`t understand how I can do it.

**Comment by oymelnyk** [comment]
Insight Object:
key = "A123"
name = "Some object"
first_attr = "INFO"
second_attr = "Additional info".

I want to get name, and first_attr.

Maybe this is a simple question, I'm sorry, as I'm a beginner

**Comment by dstengele** [comment]
Hi, don't be sorry, the documentation ist still kind of... lacking :D

I suppose, you have initialized the basic stuff as described in [the README](https://github.com/dstengele/python-insight#usage)? If so, you can do an IQL search with `object_schema.search_iql("Key = A-123")` and, as this method returns a Generator instead of a list directly, convert it to a list and get the first item:
```python
objects = [user for user in object_schema.search_iql("Key = A-123")]
insight_object = objects[0]
```

That object then holds allthe information of the Insight object and you can access its attributes via the `attributes` property which is a dict of the attribute names _as they are defined in Insight_:

```python
name = insight_object.attributes["name").value
first_attr = insight_object.attributes["first_attr").value
```

The `.value` is important as the attribute dict only holds InsightAttributeValueObjects whose actual values are requested from the REST API when needed.

**Reply by oymelnyk** [reply]
Hi,
AttributeError: 'generator' object has no attribute 'attributes'

**Reply by oymelnyk** [reply]
I found mistake.
I have 'key' and I need to use this:

obj = InsightObject (insight, 'Key')

first_attr = obj.attributes['attr_name'].value
